File: src/game/entity/models/Block.java
package game.entity.models;


import game.ui.panels.game.PitchPanel;

public abstract class Block extends Entity {
    public final static int SIZE = PitchPanel.GRID_SIZE;

    public Block(Coordinates coordinates) {
        super(coordinates);
    }

    @Override
    public int getSize() {
        return SIZE;
    }
    @Override
    public void eliminated(){
        destroy();
    }
    public void destroy(){
        despawn();
    }

    @Override
    public int getDrawPriority() {
        return 0;
    }
}

File: src/game/entity/models/Block.kt
package game.entity.models

import game.ui.panels.game.PitchPanel

abstract class Block : Entity {
    constructor(id: Long) : super(id)
    constructor(coordinates: Coordinates?) : super(coordinates)

    override val size: Int = SIZE

    override fun eliminated() = destroy()

    open fun destroy() = despawnAndNotify()

    override fun onAttackReceived(value: Int) = destroy()

    companion object {
        val SIZE = PitchPanel.GRID_SIZE
    }
}